sale_discount_total/models/discount_approval.py

```diff
@@ -37,24 +38,30 @@ class sale_discount(models.Model):
 
     @api.multi
     def action_confirm(self):
+        if self._get_forbidden_state_confirm() & set(self.mapped('state')):
+            raise UserError(_("It is not allowed to confirm an order in the following states: %s")
+                            % (", ".join(self._get_forbidden_state_confirm())))
         discnt = 0.0
         no_line = 0.0
         for order in self:
-            if self.company_id.so_double_validation == 'two_step':
+            if order.company_id.so_double_validation == 'two_step':
                 for line in order.order_line:
                     no_line += 1
                     discnt += line.discount
                 discnt = (discnt / no_line)
                 if order.company_id.so_double_validation_limit and discnt > order.company_id.so_double_validation_limit:
                     order.state = 'waiting'
-                    return True
+                    continue
```

Comment:
Please, review also this change...  
You don't have to stop confirming Sales Orders at the first one in `waiting` state... Isn't it? 🤔

Thank you.

Reply:
Thank You Byloth.